lib/src/strings.dart

```diff
@@ -94,38 +95,100 @@ String _tryYamlEncodeSingleQuoted(String string) {
 /// It is important that we ensure that [string] is free of unprintable
 /// characters by calling [_hasUnprintableCharacters] before invoking this
 /// function.
-String _tryYamlEncodeFolded(String string, int indentation, String lineEnding) {
-  String result;
-
-  final trimmedString = string.trimRight();
-  final removedPortion = string.substring(trimmedString.length);
+String _tryYamlEncodeFolded(String string, int indentSize, String lineEnding) {
+  final indent = ' ' * indentSize;
+
+  var (literalPrefix, trimmedString, stripped) = _prepareBlockString(
+    '>',
+    string,
+    indent,
+    lineEnding,
+  );
+
+  // Try folding if string has any non-empty values
+  if (trimmedString.isNotEmpty) {
+    final trimmedSplit = trimmedString.split(lineEnding);
+
+    // Apply the left indent. First line didn't have indent.
+    final first = indent + trimmedSplit[0];
+
+    if (trimmedSplit.length > 1) {
+      /// Try folding to match specification:
+      /// * https://yaml.org/spec/1.2.2/#65-line-folding
+      trimmedString =
+          trimmedSplit.foldIndexed(first, (index, previous, current) {
+        if (index == 0) return previous;
```

Comment:
Take a look at `reduceIndexed`, in which case you might not need the index and you can just use `reduce`.

> Calls combine for each element except the first.

Then you probably won't need the ` if (trimmedSplit.length > 1) {... else ....` either.

I suspect that `[foo].reduce(...) == foo` (a list of length one always reduces to the first element).